### PR TITLE
CLN/TST: indexing/multiindex/test_getitem.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
follow-on from #24452 to add section breaks and move tests. also `KeyError` checks made more explicit.

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 1]
```diff
@@ -15,6 +16,14 @@ def single_level_multiindex():
                       codes=[[0, 1, 2, 3]], names=['first'])
 
 
+@pytest.fixture
+def frame_random_data_integer_multi_index():
```
Comment (WillAyd) [inline pandas/tests/indexing/multiindex/test_loc.py]: Is this fixture duplicative of the one in conftest?

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 2]
```diff
@@ -247,3 +256,122 @@ def convert_nested_indexer(indexer_type, keys):
...
+
+@pytest.mark.parametrize('indexer, is_level1, expected_error', [
+    ([], False, None),  # empty ok
+    (['A'], False, None),
+    (['A', 'D'], False, None),
+    (['D'], False, r"\['D'\] not in index"),  # not any values found
+    (pd.IndexSlice[:, ['foo']], True, None),
+    (pd.IndexSlice[:, ['foo', 'bah']], True, None)
+])
+def test_getitem_duplicates_multiindex_missing_indexers(indexer, is_level1,
```
Comment (WillAyd) [inline pandas/tests/indexing/multiindex/test_loc.py]: I suppose it's better to rename these `get_loc` since that is in fact what they are testing

Comment (simonjayhawkins) [reply pandas/tests/indexing/multiindex/test_loc.py]: have renamed tests moved to test_loc.py to include  _loc_getitem_

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/indexing/multiindex/test_loc.py]: @WillAyd : probably yes.. it is a dataframe with random data with an integer multi-level index, so `multiindex_year_month_day_dataframe_random_data` fixture could probably be used instead.

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. ex @WillAyd commets

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 1 above)
Comment (WillAyd) [reply pandas/tests/indexing/multiindex/test_loc.py]: OK - if you want to tackle that in a follow up always good to reduce duplication

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @simonjayhawkins keep them coming!

**Pull request merged**